src/borg/repository.py

```diff
@@ -1515,7 +1523,8 @@ def iter_objects(self, segment, offset=0, include_data=False, read_data=True):
             if include_data:
                 yield tag, key, offset, data
             else:
-                yield tag, key, offset, size
+                yield tag, key, offset, size - header_size(tag)  # corresponds to len(data)
+            assert size >= 0
```

Comment:
i changed this so that it either returns the `data` (so caller could do `len(data)`) or it does not return the data, but precisely `len(data)` without any header overheads.

Reply:
if callers now want the size **with** overhead, they add `header_size(tag)` again.